public/app/features/folders/FolderLibraryPanelsPage.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+
+import { GrafanaRouteComponentProps } from '../../core/navigation/types';
+import { StoreState } from '../../types';
+import { getNavModel } from '../../core/selectors/navModel';
+import { getLoadingNav } from './state/navModel';
+import { LibraryPanelDTO } from '../library-panels/types';
+import Page from '../../core/components/Page/Page';
+import { LibraryPanelsSearch } from '../library-panels/components/LibraryPanelsSearch/LibraryPanelsSearch';
+import { OpenLibraryPanelModal } from '../library-panels/components/OpenLibraryPanelModal/OpenLibraryPanelModal';
+import { getFolderByUid } from './state/actions';
+
+export interface OwnProps extends GrafanaRouteComponentProps<{ uid: string }> {}
+
+const mapStateToProps = (state: StoreState, props: OwnProps) => {
+  const uid = props.match.params.uid;
+  return {
+    navModel: getNavModel(state.navIndex, `folder-library-panels-${uid}`, getLoadingNav(1)),
+    folderUid: uid,
+    folder: state.folder,
+  };
+};
+
+const mapDispatchToProps = {
+  getFolderByUid,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+export type Props = OwnProps & ConnectedProps<typeof connector>;
+
+export function FolderLibraryPanelsPage({ navModel, getFolderByUid, folderUid, folder }: Props): JSX.Element {
+  const [loading, setLoading] = useState(true);
+  const [selected, setSelected] = useState<LibraryPanelDTO | undefined>(undefined);
+  useEffect(() => {
+    async function loadFolder() {
+      await getFolderByUid(folderUid);
+      setLoading(false);
+    }
+    loadFolder();
+  }, [getFolderByUid, folderUid]);
```

Comment:
This looks like use case for https://github.com/streamich/react-use/blob/master/docs/useAsync.md :)

Reply:
Not a big fan, but I'll change it to useAsync.

Reply:
Fixed